platform/view/services/db/driver/sql/unversioned.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package sql
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/hyperledger-labs/fabric-smart-client/platform/view/services/db/driver"
+)
+
+type Unversioned struct {
+	base
+}
+
+func NewUnversioned(db *sql.DB, table string) *Unversioned {
+	return &Unversioned{
+		base: base{
+			db:    db,
+			table: table,
+		},
+	}
+}
+
+func (db *Unversioned) SetState(ns, key string, val []byte) error {
+	if len(val) == 0 {
+		logger.Warnf("set key [%s:%s] to nil value, will be deleted instead", ns, key)
+		return db.DeleteState(ns, key)
+	}
+	if db.txn == nil {
+		panic("programming error, writing without ongoing update")
+	}
+	logger.Debugf("set state [%s,%s]", ns, key)
+
+	val = append([]byte(nil), val...)
+
+	// Portable upsert
+	exists, err := db.exists(db.txn, ns, key)
+	if err != nil {
+		return err
+	}
+	if exists {
+		query := fmt.Sprintf("UPDATE %s SET val = $1 WHERE ns = $2 AND pkey = $3", db.table)
```

Comment:
I know this is Postgres specific, but this implementation is also related only to Postgres, so you could also use the Postgres version of UPSERT (INSERT INTO ... ON CONFLICT ...) to avoid the existence check.